crates/ingress-kafka/src/consumer_task.rs

```diff
@@ -285,15 +287,40 @@ async fn topic_partition_queue_consumption_loop(
     consumer: Arc<MessageConsumer>,
     consumer_group_id: String,
 ) -> Result<(), anyhow::Error> {
-    let mut shutdown = std::pin::pin!(cancellation_watcher());
+    let shutdown = cancellation_watcher();
+
+    let mut buffer = Vec::with_capacity(100);
+    let mut buffer_interval = tokio::time::interval(Duration::from_millis(200));
+    buffer_interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+    let mut last_offset = -1;
+
+    tokio::pin!(shutdown);
 
     loop {
         tokio::select! {
             res = topic_partition_consumer.recv() => {
                 let msg = res?;
-                let offset = msg.offset();
-                sender.send(&consumer_group_id, msg).await?;
-                consumer.store_offset(&topic, partition, offset)?;
+                last_offset = msg.offset();
+
+                buffer.push(sender.send(&consumer_group_id, msg));
```

Comment:
This `send` down the hood awaits on the bifrost append.

Reply:
After some thinking, perhaps for the time being, it just makes sense to remove this commit with the buffering. In any case this code will dramatically change when https://github.com/restatedev/restate/issues/1651 happens.

Reply:
Removed it.

Reply:
What's your take on this @AhmedSoliman ?

Reply:
I suggest you use Appender or BackgroundAppender when writing to bifrost, it's an easy change and will get you even better performance IMHO. This is adjacent to what we'll do regarding ingress -> pp communication in the future. The downside with all current options (your PR doesn't introduce the issue) is that we don't have proper back-pressure or control over faireness from partition processor so it's easy to cause overwhelm the partition processor if kafka's ingestion rate is faster than PP's processing rate.

Reply:
For now I won't do this change, we should revisit this though once we sort out how the kafka ingress behaves in the distributed setup.